config/routes.rb

```diff
@@ -8,6 +8,12 @@
     end
   end
 
+  resources :adventures do
+    collection do
+      get 'all_public', 'search'
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.